docs/content/DimensionSpecs.md

```diff
@@ -154,3 +154,207 @@ Example for the `__time` dimension:
   "function" : "function(t) { return 'Second ' + Math.floor((t % 60000) / 1000); }"
 }
 ```
+
+### Namespaced extraction function (EXPERIMENTAL)
+A namespaced extraction function is a simple key-value mapping where the key-value mappings are unique to a particular namespace.
+This can be used for re-naming values in a datasource for certain queries. To use a particular namespace, simply set the appropriate namespace for the dimExtractionFn
+```json
+    "dimExtractionFn" : {
+      "type":"namespace",
+      "namespace":"some_namespace",
+      "retainMissingValueWith":"Unknown",
+      "uniqueRenames":false
+    }
+```
+
+```json
+    "dimExtractionFn" : {
+      "type":"namespace",
+      "namespace":"some_namespace",
+      "retainMissingValue":true,
+      "uniqueRenames":false
+    }
+```
+
+A property of `retainMissingValue` and `replaceMissingValueWith` can be specified at query time to hint how to handle missing values. Setting `replaceMissingValueWith` to `""` has the same effect of setting it to `null` or omitting the property. Setting `retainMissingValue` to true will use the dimension's original value if it is not found in the namespace. The default values are `replaceMissingValueWith = null` and `retainMissingValue = false` which causes missing values to be treated as missing.
+ 
+A property of `uniqueRenames` specifies if optimizations can be used which assume there is no combining of multiple names into one. For example: If ABC123 always maps to SomeCompany, that can be optimized since it is a unique rename. But if both ABC123 and DEF456 BOTH map to SomeCompany, then that is NOT a unique rename. Setting this value and setting `retainMissingValue` to FALSE (the default) may cause undesired behavior. 
```

Comment:
I envision the most common use case would be to have one namespace essentially for one dimension of one datasource, but there is no reason it would have to be so. For example, there could be one namespace that is simply UUID -> "some string", and every dimension does its lookup against the same namespace. In such a case, there would be no sort of concept of  "unique renames" for the namespace, but the concept does exist per query.